gamecenter/api/helpers/get.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""API helpers for GET requests."""
+import iso8601
+from flask import request
+from datetime import datetime as dt
+from sqlalchemy import and_
+
+from gamecenter.api.models import Score
+
+DEFAULTS = {
+    'page_size': 5,
+    'offset': 1,
+    'sort': 'descending',
+    'start_date': dt.fromtimestamp(0),
+    'end_date': dt.utcnow()
+    }
+
+
+def construct_and_(args):
+    """Construct the 'and_' clause for a sqlalchemy query"""
+    if args['user_id'] is None:
+        if args['tag'] is None:
+            return and_(
+                Score.created_at >= args['start_date'],
+                Score.created_at <= args['end_date'])
+        else:
+            return and_(
+                Score.created_at >= args['start_date'],
+                Score.created_at <= args['end_date'],
+                Score.tag == args['tag'])
+    elif args['tag'] is None:
+        return and_(
+            Score.created_at >= args['start_date'],
+            Score.created_at <= args['end_date'],
+            Score.user_id == args['user_id'])
+    else:
+        return and_(
+            Score.created_at >= args['start_date'],
+            Score.created_at <= args['end_date'],
+            Score.tag == args['tag'],
+            Score.user_id == args['user_id'])
+
+
+def get_request_args(view_func):
```

Comment:
I knew about it, and it's not really a big deal to check them myself.. but maybe if we want to throw a 400, using webargs would be good